lib/autokey/scripting/engine.py

```diff
@@ -52,14 +52,60 @@ def get_folder(self, title: str):
                 return folder
         return None
 
+    def create_folder(self, title: str, parent_folder=None, temporary=False):
```

Comment:
You can create AutoKey folders at arbitrary paths using the GUI. So you should somehow be able to set the actual file system path. Maybe allowing `parentFolder` to be a PathLike. If it is a `str` or `pathlib.Path`, check the file system existance, try to create the directory path (recursively! `mkdir(parent=True)`) and set the path explicitly.

Not setting the path is safe though, as it will use the default location in this case.
https://github.com/autokey/autokey/blob/cd5746e7337af4895b9c7b032aa5c15473cd5f9b/lib/autokey/model.py#L438-L445

Reply:
I use a folder (sub-tree) outside of the AutoKey config for all my phrases and scripts. It makes them immune to any package modifications. This works even better now that you added code to prevent deletion of existing directories when adding a folder inside AutoKey. This has been a feature for as long as I can remember.

Reply:
I noticed that specifying the storage location is the last thing missing to make the new  `create_folder() `API function in the PR feature-complete with the GUI.



@josephj11 And unrelated to this PR: The package manager must not interfere with stuff in $HOME, as it can never know if you use self-installed applications in `~/.local` or similar. So it is always immune to the package manager regardless of location.

Reply:
Oh yeah! I realised that at some point, then forgot. Will fix

Reply:
Done